sources/Inputs/src/NetworkManager.cpp

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  NetworkManager.cpp
+ *
+ * @brief Network manager implementation file
+ *
+ */
+
+#include "NetworkManager.h"
+
+#include <DYNBusInterface.h>
+#include <DYNDataInterfaceFactory.h>
+#include <DYNLineInterface.h>
+#include <DYNNetworkInterface.h>
+#include <DYNVoltageLevelInterface.h>
+#include <boost/make_shared.hpp>
+
+namespace dfl {
+namespace inputs {
+
+NetworkManager::NetworkManager(const std::string& filepath) :
+    interface_(DYN::DataInterfaceFactory::build(DYN::DataInterfaceFactory::DATAINTERFACE_IIDM1, filepath)) {
+  buildTree();
+}
+
+void
+NetworkManager::buildTree() {
+  auto network = interface_->getNetwork();
+
+  const auto& voltageLevels = network->getVoltageLevels();
+  for (auto it_v = voltageLevels.begin(); it_v != voltageLevels.end(); ++it_v) {
+    if ((*it_v)->getVoltageLevelTopologyKind() == DYN::VoltageLevelInterface::BUS_BREAKER) {
```

Comment:
Will we add later NODE_BREAKER topology case?

Reply:
I used the algorithm in python script. I don't know if this is is still relevant

Reply:
@cossartque @marcochiaramello any insight on this?

Reply:
For now it is not necessary I guess. But at some point we might need it (maybe some automata will use a NODE_BREAKER topology). @marcochiaramello what do you think ?

Reply:
For the moment we can rely only on BUS_BREAKER topology.